File: Sixth-day-third-hw/src/main/java/com/spring/hrms/business/abstracts/JobAdvertisementService.java
package com.spring.hrms.business.abstracts;

import com.spring.hrms.core.utilities.results.DataResult;
import com.spring.hrms.core.utilities.results.Result;
import com.spring.hrms.entities.concretes.JobAdvertisement;
import com.spring.hrms.entities.dto.JobAdvDto;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface JobAdvertisementService {
    DataResult<List<JobAdvertisement>> getAll();
    DataResult<List<JobAdvertisement>> getAllSorted();
    DataResult<List<JobAdvertisement>> getAllByCompanyId(int id);
    DataResult<List<JobAdvertisement>> getAllByCompanyName(String companyName);
    DataResult<JobAdvertisement> getByJobAdvertisementId(int jobAdvertisementId);
    DataResult<List<JobAdvertisement>> getAllNotActived();

    Result add(JobAdvDto JobAdvDto);
    Result activate(int id, boolean activationStatus);
}

File: React Camp Project/camp-project/src/services/jobAdvertisementService.js
import axios from "axios"

export default class JobAdvertisementService{
    getJobAdvertisement(){
        return axios.get("http://localhost:8080/api/jobadvertisements/getAll")
    }

    getByJodAdvertisementId(id){
        return axios.get("http://localhost:8080/api/jobadvertisements/getByJobAdvertisementId?jobAdvertisementId="+id)
    }
    addJobAdvertisement(JobAdvertisement){
        return axios.post("http://localhost:8080/api/jobadvertisements/add",JobAdvertisement)
    }

    getNotActivated(){
        return axios.get("http://localhost:8080/api/jobadvertisements/getAllNotActivated")
    }

    activate(id){
        return axios.put("http://localhost:8080/api/jobadvertisements/activate?activationStatus=true&id="+id)
    }
}